central/graphql/resolvers/compliance_test.go

```diff
@@ -97,3 +107,51 @@ func (s *ComplianceResolverTestSuite) TestTruncateEmptyResults() {
 	protoassert.SlicesEqual(s.T(), testResults, truncatedResults)
 	protoassert.MapEqual(s.T(), testDomainMap, truncatedDomainMap)
 }
+
+func TestComplianceClusters(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	clusterStore := clusterMocks.NewMockDataStore(mockCtrl)
+	mainResolver := &Resolver{ClusterDataStore: clusterStore}
+
+	clusterStore.EXPECT().
+		SearchRawClusters(gomock.Any(), gomock.Any()).
+		Times(1).
+		Return(
+			[]*storage.Cluster{
+				{Id: fixtureconsts.Cluster1, Name: "Cluster 1"},
+				{Id: fixtureconsts.Cluster2, Name: "Cluster 2"},
+			},
+			nil,
+		)
+
+	identity := authnMocks.NewMockIdentity(mockCtrl)
+	identity.EXPECT().Permissions().Times(1).Return(
+		map[string]storage.Access{
+			resources.Compliance.String(): storage.Access_READ_ACCESS,
+		},
+	)
+
+	ctx := sac.WithAllAccess(context.Background())
+	ctx = authn.ContextWithIdentity(ctx, identity, t)
+
+	query := PaginatedQuery{}
+
+	fetchedClusterResolvers, err := mainResolver.ComplianceClusters(ctx, query)
+	assert.NoError(t, err)
+
+	fetchedScopeObjects := make([]*v1.ScopeObject, 0, len(fetchedClusterResolvers))
+	for _, objectResolver := range fetchedClusterResolvers {
+		if objectResolver == nil {
```

Comment:
It's not clear to me what are situations where we have an empty resolver. And what would the response for GraphQL API look like in that case?

Reply:
This was more defensive coding to avoid panic when accessing the `data` field on a nil pointer to the graphQL response.
I doubt the graphql would return nil entries in the array (at least reading the code here: https://github.com/stackrox/stackrox/pull/13080/files#diff-b81d08953ccf18ea4f1c9379d21c94d209fa40fb27ce46a8b8d9588955325f15R13324-R13333).